src/algorithms/stepindex.cpp

```diff
@@ -46,21 +56,51 @@ step_index_t::step_index_t(const PathHandleGraph& graph,
         uint64_t offset = 0;
         graph.for_each_step_in_path(
             path, [&](const step_handle_t& step) {
-                pos[step_mphf->lookup(step)] = offset;
-                offset += graph.get_length(graph.get_handle_of_step(step));
-            });
-        pos[step_mphf->lookup(graph.path_end(path))] = offset;
+				// sampling
+				if (0 == utils::modulo(graph.get_id(graph.get_handle_of_step(step)), sample_rate)) {
+					pos[step_mphf->lookup(step)] = offset;
+				}
+				offset += graph.get_length(graph.get_handle_of_step(step));
+				});
+		// sampling
+		if (0 == utils::modulo(graph.get_id(graph.get_handle_of_step(graph.path_end(path))), sample_rate)) {
+			pos[step_mphf->lookup(graph.path_end(path))] = offset;
+		}
         if (progress) {
         	building_progress_meter->increment(1);
         }
     }
 	if (progress) {
 		building_progress_meter->finish();
 	}
+	// TODO add path lengths here
 }
 
-const uint64_t& step_index_t::get_position(const step_handle_t& step) const {
-    return pos[step_mphf->lookup(step)];
+const uint64_t step_index_t::get_position(const step_handle_t& step, const PathHandleGraph& graph) const {
```

Comment:
@ekg With this function, we could print out all reference positions, if we iterate through all steps of a given reference path.